cypress/e2e/gui.cy.js

```diff
@@ -0,0 +1,145 @@
+       .should('have.value', 'Medium')
+    })
+    it('It goes back to the customers list when clicking the "Back" button *', () => {
+      cy.contains('button', "View")
+       .click()
+      cy.contains('button', "Back")	
+       .click()
+      cy.contains('p', "Below is our customer list.")
+       .should('be.visible')
+    })
+
+    it('It renders the footer with the right text and links *', () => {
+      cy.get('footer')
+       .should('be.visible')
+       .contains('Copyright 2024 - Talking About Testing')
+      cy.get('a[href="https://hotmart.com/pt-br/club/cypress-playground-ate-a-nuvem"]')
+       .should('have.text', 'Hotmart')
+      cy.get('a[href="https://udemy.com/user/walmyr" ]')
+       .should('have.text', 'Udemy')
+      cy.get('a[href="https://talkingabouttesting.com" ]')
+       .should('have.text', 'Blog')
+      cy.get('a[href="https://youtube.com/@talkingabouttesting" ]')
+       .should('have.text', 'YouTube')
+    })
+
+    it('It renders the "Hi there" greeting when no name is provided *', () => {
+      cy.contains('h2', 'Hi there')
+       .should('be.visible')
+    })
+
+    it('It renders the "Hi Joe" greeting when name is provided *', () => {
+      cy.get('input[id=name]')
+       .type('Joe')
+       cy.contains('h2', 'Hi Joe')
+       .should('be.visible')
+    })
+
+    it('It renders the header with a heading, theme`s toggle, and a text input field', () => {
+      cy.get('h1')
+       .should('be.visible')
+       .contains('EngageSphere')
+      cy.get('button[class^=ThemeToggle_button]')
+       .should('be.visible')
+      cy.get('#name')
+       .should('be.visible')
+    })
+
+    it('It opens and closes the messenger', () => {
+      cy.get('button[class^="Messenger_openCloseButton"')
+       .click()
+      cy.get('div[class^="Messenger_form"')
+       .should('be.visible')
+      cy.get('button[class^="Messenger_openCloseButton"')
+       .click()
+      cy.get('div[class^="Messenger_form"')
+       .should('not.exist')
+    })
+    it('It makes sure all messenger`s fields are mandatory and the first one is focused', () => {
+      cy.get('button[class^="Messenger_openCloseButton"')
+       .click()
+      cy.get('input[id="messenger-name"]')
+       .should('have.focus')
+       .and('have.prop', 'required', true)
+      cy.get('input[id="email"]')
+       .should('be.visible')
+       .and('have.prop', 'required', true)
+      cy.get('textarea[id="message"]')
+       .should('be.visible')
+       .and('have.prop', 'required', true)
+    })
+    it('It shows and hides a success message when successfully submitting the messenger form', () => {
+      cy.get('button[class^="Messenger_openCloseButton"')
+       .click()
+      cy.get('input[id="messenger-name"]')
+       .type('Joe')
+      cy.get('input[id="email"]')
+       .type('email@test.com')
+      cy.get('textarea[id="message"]')
+       .type('Hello there!')
+      cy.get('button[type="submit"]').click()
```

Comment:
```suggestion
      cy.get('button[type="submit"]').click()

```

Abaixo temos as verificações (_Assert_), por isso uma linha em branco aqui também.